### PR TITLE
Remove disabled-by-default servlet instrumentation

### PR DESCRIPTION
**Pull request by trask**
I don't see these being useful (even as opt-in). Sorry if we've discussed this before, you can reject this. (if it is rejected, there are a couple of misc cleanups caught up in this PR that I will break out)

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: instrumentation/servlet/servlet-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/ServletAndFilterInstrumentation.java [hunk 1]
```diff
@@ -23,19 +23,14 @@
   @Override
   public ElementMatcher<ClassLoader> classLoaderMatcher() {
     // Optimization for expensive typeMatcher.
-    return hasClassesNamed("javax.servlet.http.HttpServlet");
+    return hasClassesNamed("javax.servlet.Filter");
```
Comment (iNikem) [inline instrumentation/servlet/servlet-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/ServletAndFilterInstrumentation.java:26]: Why this change?

**Review by iNikem — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/servlet/servlet-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/ServletAndFilterInstrumentation.java (hunk 1 above)
Comment (trask) [reply instrumentation/servlet/servlet-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/ServletAndFilterInstrumentation.java:26]: mostly editorial, as it's the first one listed in the typeMatcher below

also, it's _slightly_ more specific, since Filter not introduced until servlet 2.3, though that doesn't really matter since we have `classLoaderMatcher` that already limits to 3.0+ in the InstrumentationModule
